### PR TITLE
Wip/hughsie/osx

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by hughsie** [comment]
Waiting for new libjcat and libxmlb releases before merging this is perfectly fine, I'm in no panic for this.

**Review by superm1 — COMMENTED** [review]
Rather than build directly on mac, can we cross compile for a mac target?  It's easier to fix problems that come up if we can.

**Comment by superm1** [comment]
> Waiting for new libjcat and libxmlb releases before merging this is perfectly fine, I'm in no panic for this.

Yes; please.

**Comment by hughsie** [comment]
> can we cross compile for a mac target

I don't think so; I've only seen VM based emulation. See https://stackoverflow.com/questions/40824547/docker-image-running-a-mac-os-x-installation -- there's also https://github.com/sickcodes/Docker-OSX but I'm not sure about the legality (or security) of downloading that modified OS-X image.

File: contrib/ci/build_macos.sh
Comment (hughsie) [reply contrib/ci/build_macos.sh]: Yes, all that `disable_auto_if(host_machine.system() != 'linux')` stuff works really well. I've halved the size of the `/build_macos.sh` configure section.

**Review by superm1 — DISMISSED** [review]
Lgtm. Just update subprojects to new releases and then merge.

**Review by superm1 — APPROVED** [review]
LGTM once libjcat/libxmlb are tagged.

**Pull request merged**
